accounts/abi/bind/backends/blockchain.go

```diff
@@ -90,7 +109,22 @@ func (b *BlockchainContractCaller) CallContract(ctx context.Context, call klaytn
 		return nil, err
 	}
 
-	res, err := b.callContract(call, block, state)
+	res, err := b.callContract(call, block.Header(), state)
```

Comment:
Is 'block'.Number and 'header'.number different? Still, b.callContract() can get the same header number by calling 'block'.Header() inside the function, can't it? If not, please explain.

Reply:
Yes, they are the same.

But `callContract()` had to receive Header because...
`PendingCallContract()` had only Header and no Block because...
`NewBlockchainPendingContractCaller()` had only Header and no Block because...
`Finalize()` had only `header` and no `block` [consensus/istanbul/backend/engine.go](https://github.com/klaytn/klaytn/pull/1924/files#diff-44abfea3bdbe16ce301a955732b485d1f0af2cc6c0c4cee3fbce0feb6ce051b0).

Reply:
type BlockchainContractCaller struct {
	bc            BlockChainForCaller
	pendingHeader *types.Header
	pendingState  *state.StateDB
}

I understand that the bc does not have a block body.
However, the variable name confuses me, because bc looks like a blockchain structure. wonderting why BlockChainForCaller does not have a block body.
type BlockChainForCaller interface {
	// Required by NewEVMContext
	blockchain.ChainContext

Reply:
It does have CurrentBlock, but no pending block.
Regarding KIP-103, the pending block is produced in Finalize(), which does not yet exist.
We cannot use `bc.CurrentBlock()` because `bc.CurrentBlock().Number + 1 == pendingHeader.Number`.

Reply:
I see. Thank you for the explanation.